### PR TITLE
Fix Agent Tests

### PR DESCRIPTION
**Pull request by philipktlin**
You can see this CI Build, https://msazure.visualstudio.com/One/_build/results?buildId=20580302 is passed.

### CODE REVIEW

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.E2E.Test/AgentTests.cs [hunk 1]
```diff
@@ -145,7 +143,12 @@ public async Task AgentStartsUpModules(TestConfig testConfig)
                 var dockerCommandFactory = new DockerCommandFactory(client, loggingConfig, configSource.Object, new CombinedDockerConfigProvider(Enumerable.Empty<AuthConfig>()));
                 IRuntimeInfoProvider runtimeInfoProvider = await RuntimeInfoProvider.CreateAsync(client);
                 IEnvironmentProvider environmentProvider = await DockerEnvironmentProvider.CreateAsync(runtimeInfoProvider, restartStateStore, restartManager);
-                var commandFactory = new LoggingCommandFactory(dockerCommandFactory, loggerFactory);
+
+                var logFactoryMock = new Mock<ILoggerFactory>();
```
Comment (varunpuranik) [inline edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.E2E.Test/AgentTests.cs]: Can you add some details on why this changes is needed?

Comment (philipktlin) [reply edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.E2E.Test/AgentTests.cs]: We did the same implementation in LoggingCommandFactoryTest.  Since tests will call ILoggerFactory.CreateLogger, we need to setup to return logger; otherwise the logger will throw NRE when calling Log method.

I don't know why it is working previously, but I guess updating logging NuGet package changed the behavior.

Do you want me to explicitly add comment in the code? it looks weird if only this log mocking code will have comments but not for other usages.

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.E2E.Test/AgentTests.cs [hunk 2]
```diff
@@ -145,7 +143,12 @@ public async Task AgentStartsUpModules(TestConfig testConfig)
...
+                var logMock = new Mock<ILogger<LoggingCommandFactory>>();
+                logFactoryMock.Setup(l => l.CreateLogger(It.IsAny<string>()))
+                    .Returns(logMock.Object);
+                var commandFactory = new LoggingCommandFactory(dockerCommandFactory, logFactoryMock.Object);
```
Comment (myagley) [inline edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.E2E.Test/AgentTests.cs]: Is there anywhere else we mock this `ILoggerFactory`?

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.E2E.Test/AgentTests.cs (hunk 1 above)
Comment (varunpuranik) [reply edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.E2E.Test/AgentTests.cs]: No this is good.

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.E2E.Test/AgentTests.cs (hunk 2 above)
Comment (philipktlin) [reply edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.E2E.Test/AgentTests.cs]: Yes, in LoggingCommandFactoryTest.  It is all correct in that test class.
